src/juju/jimm/JIMMV4.ts

```diff
@@ -36,14 +32,12 @@ export type RelationshipTuple = {
 
 // As typed in JIMM:
 // https://github.com/canonical/jimm/blob/c1e1642ac701bcbef2fdd8f4e347de9dcf16ac50/api/params/params.go#L324
-export type CheckRelationResponse =
-  | {
-      allowed: boolean;
-    }
-  | string;
+export type CheckRelationResponse = {
+  allowed: boolean;
+};
 
 export const isCrossModelQueryResponse = (
```

Comment:
Now that we don't get a string response we can remove this narrowing function (I'll leave a comment in the slice for what we can do instead).